edx_proctoring/api.py

```diff
@@ -1016,46 +1016,171 @@ def get_attempt_status_summary(user_id, course_id, content_id):
+        student_view_template = 'timed_exam/ready_to_submit.html'
+    elif attempt_status == ProctoredExamStudentAttemptStatus.submitted:
+        student_view_template = 'timed_exam/submitted.html'
+
+    if student_view_template:
+        template = loader.get_template(student_view_template)
+        django_context = Context(context)
+
+        allowed_time_limit_mins = attempt['allowed_time_limit_mins'] if attempt else None
+
+        if not allowed_time_limit_mins:
+            # no existing attempt, so compute the user's allowed
+            # time limit, including any accommodations
+
+            allowed_time_limit_mins = exam['time_limit_mins']
+            allowance = ProctoredExamStudentAllowance.get_allowance_for_user(
+                exam_id,
+                user_id,
+                "Additional time (minutes)"
+            )
+
+            if allowance:
+                allowed_time_limit_mins += int(allowance.value)
+
+        total_time = humanized_time(allowed_time_limit_mins)
+        progress_page_url = ''
+        try:
+            progress_page_url = reverse(
+                'courseware.views.progress',
+                args=[course_id]
+            )
+        except NoReverseMatch:
+            # we are allowing a failure here since we can't guarantee
+            # that we are running in-proc with the edx-platform LMS
+            # (for example unit tests)
+            pass
+
+        django_context.update({
+            'total_time': total_time,
+            'exam_id': exam_id,
+            'progress_page_url': progress_page_url,
+            'does_time_remain': _does_time_remain(attempt),
+            'enter_exam_endpoint': reverse('edx_proctoring.proctored_exam.attempt.collection'),
+            'change_state_url': reverse(
+                'edx_proctoring.proctored_exam.attempt',
+                args=[attempt['id']]
+            ) if attempt else '',
+        })
+        return template.render(django_context)
+
+
+def _get_proctored_exam_context(exam, attempt, course_id, exam_id):
+    """
+    Common context variables for the Proctored and Practice exams' templates.
+    """
+    attempt_time = attempt['allowed_time_limit_mins'] if attempt else exam['time_limit_mins']
+    total_time = humanized_time(attempt_time)
+    progress_page_url = ''
+    try:
+        progress_page_url = reverse(
+            'courseware.views.progress',
+            args=[course_id]
         )
-        exam = get_exam_by_content_id(course_id, content_id)
+    except NoReverseMatch:
+        # we are allowing a failure here since we can't guarantee
+        # that we are running in-proc with the edx-platform LMS
+        # (for example unit tests)
+        pass
+
+    return {
+        'platform_name': settings.PLATFORM_NAME,
+        'total_time': total_time,
+        'exam_id': exam_id,
+        'progress_page_url': progress_page_url,
+        'is_sample_attempt': True,
```

Comment:
don't hard code this value, pass as the parameter and when calling from practice_view pass the parameter to `True` and when calling from proctored_view pass it to `False`